### PR TITLE
Rails forms using form_with tag

### PR DESCRIPTION
**Pull request by felix-vondee**
In this project, we built a form with rails using the form_with tag. Other tasks completed during this project include:
- Creating a User model
- Creating a form with pure HTML as well as with rails form tags eg. <form>, form_tag and form_with
- Using validations during new user creation
- Using validations during editing details of an existing user

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @felix-vondee and @iliebabcenco  :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

#### Views

##### `new`

###### commented out `form`

* [x] uses `users_path` helper for action
* [x] uses `post` for method
* [x] uses `"UTF-8"` for ccept-charset
* [x] uses `user[<field_name>]` for input's name
* [x] uses form_authenticity_token

```html
<form action="users_path" method="post" accept-charset="UTF-8">
  <label for="user[username]">Username</label>
  <input name="user[username]" type="text" />
  <input
    type="hidden"
    name="authenticity_token"
    value="<%= form_authenticity_token %>"
  />
  <input type="submit" value="Submit" />
</form>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

###### commented out `form_tag`

* [x] uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`

```html
<%= form_tag(users_path) do %> <%= label_tag 'username' %> <%= text_field_tag
'username' %> <%= label_tag 'email' %> <%= email_field_tag 'email' %> <%=
label_tag 'password' %> <%= password_field_tag 'password' %> <%= submit_tag %>
<% end %>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))


### Add the commented out form and submit for another review.



## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  <br>
+  <br>
+  <%= label_tag(:password, "Password") %>
+  <%= form.password_field :password %>
+  <br>
+  <br>
+  <%= form.submit "Create" %>
+<% end %>
+<!--<hr>
+<form method="post" action="/users" accept-charset="UTF-8" data-remote="true">
```
Comment (descholar-ceo) [inline app/views/users/new.html.erb]: [required] As described in the first review, the action should be `action="users_path"` not ~action="/users"~

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <%= label_tag 'email' %>
+  <%= email_field_tag 'email' %>
+  <br>
+    <br>
+    <%= label_tag 'password' %>
+  <%= password_field_tag 'password' %>
+  <br>
+    <br>
+    <%= submit_tag %>
+  <% end %> -->
```
Comment (descholar-ceo) [inline app/views/users/new.html.erb]: [Optional] This form_tag should be commented out

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @felix-vondee and @iliebabcenco  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's because this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] There are still requested changes to fix out, kindly checkout my inline comments
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by descholar-ceo — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @felix-vondee and @iliebabcenco  , your project is approved :tada: :tada: :tada: 
You have implemented all minimum requirements for this project to mark it as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Pull request merged**